src/include/common/types/timestamp.hpp

```diff
@@ -10,9 +10,60 @@
 
 #include "common/common.hpp"
 
+#include <chrono>  // chrono::system_clock
+#include <ctime>   // localtime
+#include <iomanip> // put_time
+#include <sstream> // stringstream
+#include <string>  // string
+
 namespace duckdb {
 
-//! The Date class is a static class that holds helper functions for the Timestamp
+constexpr const double DAYS_PER_YEAR = 365.25; //! assumes leap year every four years
+constexpr const size_t MONTHS_PER_YEAR = 12;
+constexpr const size_t DAYS_PER_MONTH = 30; //! assumes exactly 30 days per month (ISO 8601 suggest 30 days)
+constexpr const size_t HOURS_PER_DAY = 24;  //! assume no daylight savings time changes
+constexpr const size_t STD_TIMESTAMP_LENGTH = 19;
+constexpr const size_t MAX_TIMESTAMP_LENGTH = 23;
+constexpr const char *DEFAULT_TIME = "00:00:00";
+
+/*
+ *	This doesn't adjust for uneven daylight savings time intervals or leap
+ *	seconds, and it crudely estimates leap years.  A more accurate value
+ *	for days per years is 365.2422.
+ */
+constexpr const size_t SECS_PER_YEAR = (36525 * 864); /* avoid floating-point computation */
+constexpr const size_t SECS_PER_DAY = 86400;
+constexpr const size_t SECS_PER_HOUR = 3600;
+constexpr const size_t SECS_PER_MINUTE = 60;
+constexpr const size_t MINS_PER_H = 60;
+constexpr const int64_t MSECS_PER_DAY = 8640000;
+constexpr const int64_t MSECS_PER_HOUR = 360000;
+constexpr const int64_t MSECS_PER_MINUTE = 60000;
+constexpr const int64_t MSECS_PER_SEC = 1000;
+
+// Used to check amount of days per month in common year and leap year
+constexpr int days_per_month[2][13] = {{31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31, 0},
+                                       {31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31, 0}};
+constexpr bool isleap(int16_t year) {
+	return (((year) % 4) == 0 && (((year) % 100) != 0 || ((year) % 400) == 0));
+}
+
+struct Interval {
+	int64_t time;
+	int32_t days;   //! days, after time for alignment
+	int32_t months; //! months after time for alignment
+};
+
+struct timestamp_struct {
+	int16_t year;
```

Comment:
Year should be an int32_t. Years bigger than 2^16 are supported (e.g. `SELECT DATE '100000-01-01';`). Please add a testcase with this as well.

Reply:
I just realized that this is broken with dates in the C API as well, as SELECT DATE '100000-01-01' displays an overflowed/underflowed date. This should be fixed as well, but is not in the scope of this PR and hence I will fix it myself.

Reply:
Ok. I'm gonna change the data type, then.